hw/ip/pwm/doc/_index.md

```diff
@@ -0,0 +1,313 @@
+The resolution can be also reduced down as low as 1-bit, to allow for more frequent pulses when connected to a low-frequency, low-power bus.
+
+This IP only controls the *timing* of the PWM drive signal.
+The drive *current* must be managed by the system designer by including the appropriate off-chip circuitry.
+For instance, in the LED-drive use-case, various off-chip solutions exist for controlling the current, such as selection of an appropriate series resistance, or the use of an external fixed-current LED drive IC.
+Some limited possibilities for current control may also exist in a top-level ASIC design.
+However, such configuration options are outside the scope of this document.
+
+In order to support a variety of drive configurations, the polarity of each channel may be inverted.
+
+The PWM IP is also suitable for driving other outputs, such as servo motors.
+In some applications, the user may wish to stagger outputs to limit the overall magnitude of current spikes in the complete system.
+Thus each output also has a phase control register, with up to 16-bit resolution.
+In security applications it is the system-designer's responsibility to ensure that improper configuration of the phase control registers does not pose a security risk to the overall system (due to, for instance, a denial of service attack through the PWM configuration).
+
+The PWM IP provides a hardware-controlled blink feature, which can periodically toggle the output between two states with separate duty cycles.
+This feature can be used to significantly reduce the output duty cycle (blink-off), or, in the case of tri-color LEDs, toggle the apparent LED color between the two settings.
+This feature is enabled on a per-channel basis, and for each blinking channel, the blink period and blink duty cycle are programmed in terms of the number of pulses generated in either state.
+
+As a variant of the blink feature, the output duty cycle can also be programmed to linearly increase and decrease in time.
+In this "heartbeat" mode, the duty cycle increments by a programmable amount after a programmable number of pulses, starting at some firmware-selected initial duty cycle.
+Once the internal duty cycle reaches the target value, the internal duty cycle begins to decrement until it returns to the initial value, at which point the cycle repeats until heartbeat mode is disabled.
+
+# Theory of Operations
+
+## Block Diagram
+
+![](pwm_block_diagram.svg)
+
+## Hardware Interfaces
+
+{{< hwcfg "hw/ip/pwm/data/pwm.hjson" >}}
+
+## Design Details
+
+### Phase and Duty Cycle Representation
+
+The PWM IP creates series of pulses with the desired on-off duty cycle.
+The duty cycle, DC, is typically expressed a fraction of pulse duration, <i>d</i>, over the period between pulses, <i>T</i>:
+$$DC\equiv d/T$$.
+Since 0&lt;<i>d</i>&lt;<i>T</i>, the duty cycle generally ranges from 0 to 1.
```

Comment:
This sounds like it might sometimes not be in the range`[0, 1]`, which I think is false. Maybe just remove "generally"?

Reply:
My point was that the user configures the duty cycle (which, in turn, affects `d`) by setting some register and the result is always in the range `[0, 1]`. The "generally" makes it look like it's not always true, which might be a bit confusing.

Reply:
Oh! You want to make the statement more firm, not softer. Sorry, I misunderstood you. Sure, I will remove "generally". Makes sense to me now.